cpp/src/arrow/compute/kernel.h

```diff
@@ -78,6 +79,14 @@ struct ARROW_EXPORT Datum {
 
   Datum(const Datum& other) noexcept { this->value = other.value; }
 
+  // Define move constructor and move assignment, for better performance
```

Comment:
can't you do the equivalent with `= default`?

Reply:
I had thought so, too -- don't structs have a default move ctor and assignment without explicit declaration?